services/QuillLMS/client/app/bundles/Staff/components/comprehension/semanticRules/model.tsx

```diff
@@ -1,29 +1,44 @@
 import * as React from "react";
-import { useQuery } from 'react-query';
-import { withRouter, Link } from 'react-router-dom';
+import { useQuery, queryCache } from 'react-query';
+import { withRouter, Link, useHistory } from 'react-router-dom';
 import { EditorState, ContentState } from 'draft-js';
 
-import { fetchModel } from '../../../utils/comprehension/modelAPIs';
+import { fetchModel, updateModel } from '../../../utils/comprehension/modelAPIs';
 import { DataTable, Spinner, TextEditor } from '../../../../Shared/index';
+import * as request from 'request';
 
 const Model = ({ match }) => {
   const { params } = match;
   const { activityId, modelId } = params;
 
-  // const [modelNotes, setModelNotes] = React.useState<string>('');
   const [errors, setErrors] = React.useState<object>({});
+  let history = useHistory();
 
   // cache ruleSets data for handling rule suborder
   const { data: modelData } = useQuery({
     queryKey: [`model-${modelId}`, modelId],
     queryFn: fetchModel
   });
 
-  // function handleSetModelNotes(text: string){ setModelNotes(text) };
+  const initialNoteValue = modelData ? modelData.model.notes : '';
+  const [modelNotes, setModelNotes] = React.useState<string>(initialNoteValue);
 
-  // function onHandleUpdateModel() {
+  function handleSetModelNotes(text: string){ setModelNotes(text) };
 
-  // }
+  function onHandleUpdateModel() {
+    updateModel(modelId, modelNotes).then((response) => {
+      const { error } = response;
+      
+      if(error) {
+        const updatedErrors = {};
+        updatedErrors['Model Submission Error'] = error;
+        setErrors(updatedErrors);
+      } else {
+        queryCache.refetchQueries(`model-${modelId}`);
```

Comment:
@emilia-friedberg I patterned this code off of https://github.com/empirical-org/Empirical-Core/blob/develop/services/QuillLMS/client/app/bundles/Staff/components/comprehension/semanticRules/semanticLabelForm.tsx#L146

However, `refetchQueries` does not seem to be refetching the desired query:  once the user is navigated to the index page and clicks "View Model" again, the old note state is displayed. Only by refreshing the Model View page will the correct updated note be displayed.

Reply:
I think @eadams17 had a relevant realization about `refetchQueries` recently, he can explain better than I can if you ask him!

Reply:
yeah I think you need to update this line to be `queryCache.refetchQueries([`model-${modelId}`, modelId])` and it should work

Reply:
so `refetchQueries` has been phased out. The docs show that `queryCache.clear()` is the new alternative: https://react-query.tanstack.com/reference/QueryCache

Reply:
Thanks; the second option worked.